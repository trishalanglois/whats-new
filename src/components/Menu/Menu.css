/* .menu {
  grid-area: menu;
} */

.menu {
  width: 20%;
  float: left;
  height: 100%;
}
