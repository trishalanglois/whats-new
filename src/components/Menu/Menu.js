import React from 'react';
import './Menu.css';
import youAreHere from '../../images/you-are-here.png';
import entertainmentIcon from '../../images/entertainment.png';
import healthIcon from '../../images/health.png';
import scienceIcon from '../../images/science.png';
import technologyIcon from '../../images/technology.png';


const Menu = (props) => (
  <nav className = 'menu'>
    <h1 id='logo'>News Your Own <span id='new-feature-word'>Adventure</span></h1>
    <button onClick = {() => props.changeNews(props.local)} className='menu-button'><img className='menu-button-icon' src={youAreHere} />  Local News</button>
    <button onClick = {() => props.changeNews(props.technology)}className='menu-button'><img className='menu-button-icon' src={technologyIcon} /> Technology</button>
    <button onClick = {() => props.changeNews(props.entertainment)}className='menu-button'><img className='menu-button-icon' src={entertainmentIcon} /> Entertainment</button>
    <button onClick = {() => props.changeNews(props.science)}className='menu-button'><img className='menu-button-icon' src={scienceIcon} /> Science</button>
    <button onClick = {() => props.changeNews(props.health)}className='menu-button'><img className='menu-button-icon' src={healthIcon} /> Health</button>
  </nav>
);



export default Menu
