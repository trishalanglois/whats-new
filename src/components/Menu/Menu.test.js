import React from 'react';
import Menu from './Menu';
import { shallow } from 'enzyme';

describe('Menu', () => {
  it('should correctly match the snapshot with all the data passed correctly', () => {
    const wrapper = shallow(
      <Menu
        changeNews={jest.fn()}
      />)
      expect(wrapper).toMatchSnapshot();
  })

  it('should call the changeNews prop with the local argument when the Local button is clicked', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Menu
        changeNews={mockClick}
      /> );
    wrapper.find('#local-button').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  })

  it('should call the changeNews prop with the technology argument when the Technology button is clicked', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Menu
        changeNews={mockClick}
      /> );
    wrapper.find('#technology-button').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  })

  it('should call the changeNews prop with the entertainment argument when the Entertainment button is clicked', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Menu
        changeNews={mockClick}
      /> );
    wrapper.find('#entertainment-button').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  })

  it('should call the changeNews prop with the science argument when the Science button is clicked', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Menu
        changeNews={mockClick}
      /> );
    wrapper.find('#science-button').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  })

  it('should call the changeNews prop with the health argument when the Health button is clicked', () => {
    const mockClick = jest.fn();
    const wrapper = shallow(<Menu
        changeNews={mockClick}
      /> );
    wrapper.find('#health-button').simulate('click');
    expect(mockClick).toHaveBeenCalled();
  })
})
