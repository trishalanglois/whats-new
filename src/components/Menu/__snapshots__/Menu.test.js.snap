// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should correctly match the snapshot with all the data passed correctly 1`] = `
<nav
  className="menu"
>
  <h1
    id="logo"
  >
    News Your Own 
    <span
      id="new-feature-word"
    >
      Adventure
    </span>
  </h1>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <img
      className="menu-button-icon"
      src="you-are-here.png"
    />
      Local News
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <img
      className="menu-button-icon"
      src="technology.png"
    />
     Technology
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <img
      className="menu-button-icon"
      src="entertainment.png"
    />
     Entertainment
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <img
      className="menu-button-icon"
      src="science.png"
    />
     Science
  </button>
  <button
    className="menu-button"
    onClick={[Function]}
  >
    <img
      className="menu-button-icon"
      src="health.png"
    />
     Health
  </button>
</nav>
`;
